pkg/utils/data_generator.go

```diff
@@ -0,0 +1,52 @@
+package utils
+
+import (
+	"math/rand"
+
+	"github.com/gonum/matrix/mat64"
+)
+
+// Creates random data samples clustered in the given number of clusters.
+// The samples are located in a hypercube given by the max and min parameters.
+// The clusters are hypercubes (not hyperspheres) around randomly picked cluster centres.
+// There is the same number of data samples in each cluster (rows/clusters)
+// rows - how many data rows to generate
+// cols - dimension of data
+// clusters - how many clusters
+// max,min - maximum and minimum coordinates (applies to all dimensions)
+// vari - this is the interval where the data will be located around the cluster centres.
```

Comment:
damn, this var name is not very straightforward :-/ Neighbourhood or something like that would suit it better? What do you think? It would make it more consistent with SOM itself.